### ISSUE TITLE
bulkrename doesn't work if open txt file in a new terminal

### ISSUE THREAD

**Issue by Xunius** [ranger/ranger#880]
If I set ranger to open txt files in a new terminal by changing the relevant line in `rifle.conf`:
```
mime ^text,  label editor = $EDITOR -- "$@"
```
to
```
mime ^text,  flag t, label editor = $EDITOR -- "$@"
```
Then when doing a bulkrename, the newly opened text editor contains no file names in it.

#### ISSUE TYPE
- Bug report

#### RUNTIME ENVIRONMENT

- Operating system and version: Manjaro 17.0.1
- Terminal emulator and version: Gnome-terminal 3.22.1
- Python version: 3.6.0
- Ranger version/commit: ranger-stable 1.8.1
- Locale: en_US.utf-8

#### EXPECTED BEHAVIOR

In ranger, select multiple files, then `:bulkrename` opens the editor with selected file names.

#### CURRENT BEHAVIOR

If set to open editor in a new terminal, editor (`vim` in my case) opens but with nothing in it.

**Comment by Vifon** [comment]
It's happening because the `t` flag causes the spawned process to fork, so ranger doesn't wait for it to exit and removes the file with the `:bulkrename` commands right away, before the spawned (and forked) editor can open it. It is the expected behavior in general.

What is not expected and needs fixing is that ranger uses the general text editor rifle rule for `:bulkrename`. If anyone has a good idea how to handle this corner case, feel free to comment.

**Comment by magnetophon** [comment]
> What is not expected and needs fixing is that ranger uses the general text editor rifle rule for :bulkrename

I'm having a similar problem, and I'm not sure if I should open a separate issue for that:
In ``rifle.conf``, I've set the text editor to explicitly fork, so I can edit text and not block ranger.
This of course breaks :bulkrename, and possibly other commands.

Should I create a separate issue for that?

**Comment by vicmortelmans** [comment]
I have the same problem when I set $EDITOR to be gvim in ~/.bashrc. When I make it vim, bulkrename is working!

**Comment by tigerjack** [comment]
Same problem here. As a workaround I simply removed any flag associated to the default editor in `rifle.conf`.

**Comment by cpchan** [comment]
Same problem here. I ended up copying the function to commands.py and hard coding the editor.

**Comment by chu4ng** [comment]
I often get it. And I don't like solution with vidir.
So I add new entry in rifle with another label and edit "app" var to this label. 
commands.py +/class\ bulkrename/
`self.fm.execute_file([File(listpath)], app='bulk_editor')
`

**Comment by camoz** [comment]
What about making `:bulkrename` just not use rifle, and have it always open `$VISUAL`/`$EDITOR` in the current terminal? This way users could still specify their favorite text editor, but it would not be possible to open the editor in a separate window/process for `:bulkrename`, like with rifle's `t` flag. Seems like a straightforward fix to me.

**Comment by cpchan** [comment]
> What about making `:bulkrename` just not use rifle, and have it always open `$VISUAL`/`$EDITOR` in the current terminal? This way users could still specify their favorite text editor, but it would not be possible to open the editor in a separate window/process for `:bulkrename`, like with rifle's `t` flag. Seems like a straightforward fix to me.

This will not work in all cases. The editor defined in $VISUAL/$EDITOR can fork too. For example I have mine setup to passed the "-n" flag to "emacsclient". This will cause the connection to the calling terminal to terminate and open up the file in a new frame in Emacs under X.

**Comment by toonn** [comment]
In the short term the best solution is to have "label editor" on a non-forking editor and then have another rule that matches text files above that *without* the editor label.

**Comment by SamDc73** [comment]
are there any workaround for now ?

**Comment by Ginner** [comment]
Yeah. Following toonn's advice above, I've got the following in `rifle.conf`:
```
mime ^text  = nohup $TERMINAL  -- ${VISUAL:-$EDITOR} "$@" > /dev/null &
mime ^text,  label editor = ${VISUAL:-$EDITOR} -- "$@"
```
I.e. "label editor" as a non-forking editor and a rule before that matches text files. It works like a charm, both `bulkrename` and forking an editor when editing a text file.

**Comment by toonn** [comment]
I'm considering adding a `bulkrename` label. We can then add new Rifle rules with that label and explain that it should only be used with processes that do not fork. Potentially we can even refactor bulkrename so it can be called from the command line, something like `ranger --bulkrename -- <file with paths>`, or into its own script. Then we can have rules like these:

```
label bulkrename = vidir < "$1"
label bulkrename = ranger --bulkrename -- "$1"
```

The best part about this change would be that we can support alternative implementations of the feature (that can be more robust, it's way harder to accidentally delete files with vidir than with bulkrename).
